rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/HostsFileBasedInfrastructureManager.java

```diff
@@ -384,211 +382,180 @@ private int getRetriesLeft(int retries) {
-        persistedInfraVariables.put(PN_TIMEOUT_KEY, new HashMap<String, Boolean>());
-        persistedInfraVariables.put(HOST_PER_NODE_URL_KEY, new HashMap<String, InetAddress>());
     }
 
     /**
      * Launch the node on the host passed as parameter
-     * @param host The host on which one the node will be started
+     * @param hostTracker The host on which one the node will be started
      * @param nbNodes number of nodes to deploy
      * @param depNodeURLs list of deploying or lost nodes urls created 
      * @throws RMException If the node hasn't been started. Very important to take care of that
      * in implementations to keep the infrastructure in a coherent state.
      */
-    protected abstract void startNodeImpl(InetAddress host, int nbNodes, List<String> depNodeURLs) throws RMException;
+    protected abstract void startNodeImpl(HostTracker hostTracker, int nbNodes, List<String> depNodeURLs)
+            throws RMException;
 
     /**
      * Kills the node passed as parameter
      * @param node The node to kill
-     * @param host
+     * @param host The host of the node
      * @throws RMException if a problem occurred while removing
      */
     protected abstract void killNodeImpl(Node node, InetAddress host) throws RMException;
 
     /**
-     * Removes a node from the registered nodes and adds one node removed to the removed host map.
-     * In case all nodes relative to this host were removed, the JVM is killed.
-     *
-     * @param node the node to remove
-     *
-     * @return the {@see InetAddress} of the host of the removed node
+     * Utility class to encapsulate node name building and node name parsing.
+     * The node name involves the host that was written in the host file, and
+     * that si used as a identifying key to track the node afterwards.
      */
-    private InetAddress removeNodeAndReturnHost(final String nodeName, final String nodeUrl, final Node node) {
-        return setPersistedInfraVariable(new PersistedInfraVariablesHandler<InetAddress>() {
-            @Override
-            public InetAddress handle() {
-                InetAddress host = getRegisteredNodes().remove(nodeName);
-                if (host != null) {
-                    logger.debug("Removing node " + nodeUrl + " from " + this.getClass().getSimpleName());
-                    addRemovedHost(host);
-                    if (!getRegisteredNodes().containsValue(host) && node != null) {
-                        try {
-                            killNodeImpl(node, host);
-                        } catch (Exception e) {
-                            logger.trace("An exception occurred during node kill", e);
-                        }
-                    }
-                }
-                return host;
-            }
-        });
-    }
+    protected class NodeNameBuilder implements Serializable {
+
+        private static final String SSH_NODE_NAME_PREFIX = "SSH";
```

Comment:
If the builder is defined in HostsFileBasedInfrastructure, its prefix is not necessary SSH, it could be added as an extra parameter

Reply:
Yes you are right. I changed it by using the name of the concrete class instead of "SSH".